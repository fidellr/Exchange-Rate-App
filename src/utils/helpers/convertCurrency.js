const convertCurrencyName = (name) => {
  switch (name) {
    case 'AUD':
      return 'Australian Dollar';
    case 'BGN':
      return 'Bulgarian Lev';
    case 'BRL':
      return 'Brazilian Real';
    case 'CAD':
      return 'Canadian Dollar';
    case 'CHF':
      return 'Swiss Franc';
    case 'CNY':
      return 'Chinese Yuan';
    case 'CZK':
      return 'Czech Koruna';
    case 'DKK':
      return 'Danish Krone';
    case 'EUR':
      return 'Euro';
    case 'GBP':
      return 'British Pound';
    case 'HKD':
      return 'Hong Kong Dollar';
    case 'HRK':
      return 'Croatian Kuna';
    case 'HUF':
      return 'Hungarian Forint';
    case 'IDR':
      return 'Indonesian Rupiah';
    case 'ILS':
      return 'Israeli New Shekel';
    case 'INR':
      return 'Indian Rupee';
    case 'ISK':
      return 'Icelandic Króna';
    case 'JPY':
      return 'Japanese Yen';
    case 'KRW':
      return 'South Korean Won';
    case 'MXN':
      return 'Mexican Peso';
    case 'MYR':
      return 'Malaysian Ringgit';
    case 'NOK':
      return 'Norwegian Krone';
    case 'NZD':
      return 'New Zealand Dollar';
    case 'PHP':
      return 'Philipine Peso';
    case 'PLN':
      return 'Polish Złoty';
    case 'RON':
      return 'Romanian Leu';
    case 'RUB':
      return 'Russian Ruble';
    case 'SEK':
      return 'Swedish Krona';
    case 'SGD':
      return 'Singapore Dollar';
    case 'THB':
      return 'Thai Baht';
    case 'TRY':
      return 'Turkish Lira';
    case 'USD':
      return 'United States Dollar';
    case 'ZAR':
      return 'South African Rand';
    default: return '';
  }
}

export {
  convertCurrencyName,
}