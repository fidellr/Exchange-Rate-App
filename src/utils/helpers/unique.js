const isExist = (originArr, key, targetValue) => {
  let exist = false;
  for (let i=0; i < originArr.length; i++) {
    if (originArr[i][key] === targetValue) {
      exist = true;
    }
  }

  return exist; 
}

export {
  isExist,
};
