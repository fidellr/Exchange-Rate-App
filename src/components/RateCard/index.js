import React from 'react';
import { convertCurrencyName } from '../../utils/helpers/convertCurrency';
import RemoveIcon from '../../static/icons/remove.svg';
import Card from '../Card';

const rateDetailClasses = 'u-margin0 u-fontSize16';
export default ({
  onRemove,
  baseCurrency,
  currentBaseValue,
  rateName,
  rateNumber,
}) => (
    <Card className="rateCard-wrapper u-flex u-justifyContentSpaceBetween u-backgroundColorWhite">
      <div className="rateCard-details u-sizeFullWidth u-padding4">
        <div className="rateCard-detailHeader u-flex u-justifyContentSpaceBetween">
          <p className={rateDetailClasses}>{rateName}</p>
          <p className={rateDetailClasses}>{(rateNumber * currentBaseValue).toFixed(4)}</p>
        </div>
        <p className="u-margin0 u-fontSize12 u-fontWeight7 u-marginVertical4">{`${rateName}-${convertCurrencyName(rateName)}`}</p>
        <p className="u-margin0 u-fontSize12 u-fontWeight7 u-marginVertical4">1 {baseCurrency} = {rateName} {rateNumber}</p>
      </div>
      <div className="rateCard-button u-width50 u-borderLeftColorGray1 u-relative">
        <div
          className="deleteButton u-cursorPointer u-absolute u-top0 u-bottom0 u-right0 u-left0 u-backgroundSizeCover u-backgroundNoRepeat u-size24 u-marginAuto"
          onClick={onRemove}
          style={{ backgroundImage: `url(${RemoveIcon})` }}
        />
      </div>
    </Card>
  );
