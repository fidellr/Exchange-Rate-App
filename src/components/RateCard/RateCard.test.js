import React from 'react';
import { shallow } from 'enzyme';
import RateCard from './index';
import { convertCurrencyName } from '../../utils/helpers/convertCurrency';

describe('RateCard', () => {
  it('should render converted abbreviation of currency that passed through props and the abbreviation too', () => {
    const component = shallow(<RateCard rateName="SGD" />);
    const convertedAbbrev = convertCurrencyName('SGD');
    expect(component.childAt(0).childAt(1).text()).toContain(convertedAbbrev);
    expect(component).toMatchSnapshot();
  });
});