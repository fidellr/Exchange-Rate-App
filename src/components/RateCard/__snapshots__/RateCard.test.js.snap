// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateCard should render converted abbreviation of currency that passed through props and the abbreviation too 1`] = `
<_default
  className="rateCard-wrapper u-flex u-justifyContentSpaceBetween u-backgroundColorWhite"
>
  <div
    className="rateCard-details u-sizeFullWidth u-padding4"
  >
    <div
      className="rateCard-detailHeader u-flex u-justifyContentSpaceBetween"
    >
      <p
        className="u-margin0 u-fontSize16"
      >
        SGD
      </p>
      <p
        className="u-margin0 u-fontSize16"
      >
        NaN
      </p>
    </div>
    <p
      className="u-margin0 u-fontSize12 u-fontWeight7 u-marginVertical4"
    >
      SGD-Singapore Dollar
    </p>
    <p
      className="u-margin0 u-fontSize12 u-fontWeight7 u-marginVertical4"
    >
      1 
       = 
      SGD
       
    </p>
  </div>
  <div
    className="rateCard-button u-width50 u-borderLeftColorGray1 u-relative"
  >
    <div
      className="deleteButton u-cursorPointer u-absolute u-top0 u-bottom0 u-right0 u-left0 u-backgroundSizeCover u-backgroundNoRepeat u-size24 u-marginAuto"
      style={
        Object {
          "backgroundImage": "url(remove.svg)",
        }
      }
    />
  </div>
</_default>
`;
